CI improvements

ffromani opened the pull request:
this script is kni-specific, hence belongs on hack-kni.
It was originally added on hack/ because it predated the hack/hack-kni
split.

Improve the verify-commits.sh script to be more precise in checking (just check the subject line) and to have better reporting
Fix the `unit-test-quick.sh` to be consistent with go.mod

Signed-off-by: Francesco Romani <fromani@redhat.com>

cynepco3hahue commented:
I am unsure the title is fully correct, in general:
1. You need informers to get updates regarding relevant objects and call event handlers
2. You need listers to get relevant objects from the cache instead of the API.

fromanirh commented:
> I am unsure the title is fully correct, in general:
> 
>     1. You need informers to get updates regarding relevant objects and call event handlers
> 
>     2. You need listers to get relevant objects from the cache instead of the API.

This is why it's WIP ;) The idea is indeed to reshuffle the components client-go gives us, and how we use them.

The main benefit would to be overcome the need to explicitely list the allowed namespaces. With the informer-first approach which I'm experimenting with in this PR, it's pretty trivial to watch for all the objects in the cluster, while we still have the option to watch only specific namespaces.

fromanirh commented:
no longer needed as we move towards cluster-scoped CRD

ffromani committed:
[KNI] move verify-commit.sh helper in hack-kni

this script is kni-specific, hence belongs on hack-kni.
It was originally added on hack/ because it predated the hack/hack-kni
split.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
[KNI] verify-commit.sh: consistent logging

make sure to log the same commits we will check right
after.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
[KNI] gh actions: don't setup go when checking commits

We don't need to setup golang to verify the
commit messages, so let's not do it.

Signed-off-by: Francesco Romani <fromani@redhat.com>

Tal-or commented on hack-kni/verify-commits.sh:
```diff
@@ -48,7 +48,7 @@ git log --oneline "$latest_upstream_commit"..HEAD
 
 # list commits
-for commitish in $(git log --oneline --no-merges "$latest_upstream_commit"..HEAD | cut -d' ' -f 1); do
+for commitish in $(git log --oneline --no-merges "$latest_upstream_commit"..HEAD~ | cut -d' ' -f 1); do
```
The ```--no-merges``` flag should take care of that, so ```HEAD~``` won't cause skipping the check of the last commit?

ffromani replied:
yes, this is just wrong. The fix (if needed) is elsewhere

Tal-or commented on .github/workflows/ci.yaml:
```diff
@@ -14,10 +14,5 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: set up golang
```
The commit is missing the [KNI] prefix but yet the CI passed. I really need to revisit the script

ffromani replied:
yup, this is what I was testing.

ffromani committed:
[KNI] verify-commit: more precise check

check just the subject line, because this is what
we actually care about, and improve the reporting.

Signed-off-by: Francesco Romani <fromani@redhat.com>

Tal-or commented:
I think the problem is here:
https://github.com/openshift-kni/scheduler-plugins/blob/master/.github/hooks/commit-msg#L6
please change the line to:
``` if [[ "$(head -1 $cmtmsg)" != \[KNI\]* ]]; then```

I recall I fixed that when it was under my org, but maybe I forgot to push it or something

Tal-or commented:
@fromanirh for the ci/prow/ci-unit the fix is: https://github.com/openshift-kni/scheduler-plugins/pull/10

ffromani committed:
[KNI] rename CI lane

in this lane we check the commit messages, so rename accordingly

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
[KNI] make unit-test-quick consistent with go.mod

We need to use the go module we declare;
we declare the same go.mod as u/s because
it seems the way to go (xref: openshift/kubernetes)

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
> @fromanirh for the ci/prow/ci-unit the fix is: #10

thanks, actually not sure we should use the go module name or the relative path.
Going with go module name mostly because this is what `hack/unit-test.sh` is doing.

Tal-or commented:
/lgtm
/approve
